File: Backend/models/userModel.js
const db = require('../config/db');

async function getUserWithCostCenters(uid) {
  const [userRows] = await db.execute('SELECT * FROM users WHERE uid = ?', [uid]);
  if (userRows.length === 0) throw new Error('User not found');

  const user = userRows[0];
  const [costCenters] = await db.execute(
    'SELECT cost_center_id, cost_center_name, regon FROM cost_centers WHERE user_id = ?',
    [user.id]
  );

  return {
    id: user.id,
    name: user.name,
    role: user.role,
    position: user.position,
    cost_centers: costCenters,
  };
}

module.exports = {
  getUserWithCostCenters,
};

File: lib/models/user_model.dart
class UserModel {
  final int id;
  final String name;
  final String role;
  final String position;
  final List<CostCenter> costCenters;

  UserModel({
    required this.id,
    required this.name,
    required this.role,
    required this.position,
    required this.costCenters,
  });

  factory UserModel.fromJson(Map<String, dynamic> json) {
    var list = json['cost_centers'] as List;
    List<CostCenter> centers = list.map((e) => CostCenter.fromJson(e)).toList();
    return UserModel(
      id: json['id'],
      name: json['name'],
      role: json['role'],
      position: json['position'],
      costCenters: centers,
    );
  }
}

class CostCenter {
  final int id;
  final String name;
  final String cost;

  CostCenter( {required this.id, required this.name, required this.cost,});

  factory CostCenter.fromJson(Map<String, dynamic> json) {
    return CostCenter(id: json['cost_center_id'], name: json['cost_center_name'], cost:json['regon']);
  }
}
